src/prompt/variable.rs

```diff
@@ -146,27 +441,82 @@ pub struct Json;
 
 impl VariableInput for Json {
     fn type_name(&self) -> &str { "json" }
-    fn parse(&self, input: &str) -> Result<Value, Error> {
-        match serde_json::from_str::<serde_json::Value>(input) {
-            Err(e) if e.classify()  == serde_json::error::Category::Eof
-            => Err(Error::Incomplete),
-            Err(e) => Err(no_pos_err(e)),
-            Ok(_) => Ok(Value::Json(
-                model::Json::new_unchecked(input.into())
-            )),
-        }
+    fn parse<'a>(&self, input: &'a str, _flags: InputFlags) -> ParseResult<'a> {
+        // treat as string, and then validate the strings content
```

Comment:
This is not right; the json might not be a string.

This one is kind of a pain in the ass. serde does not seem to care about making this easy.
I think to do it we'd want to use https://docs.rs/serde_json/latest/serde_json/struct.StreamDeserializer.html, where we can parse one value out of the stream and then use `byte_offset` to figure out how much to advance the parser

Another thing we *could* do is requiring it be quoted in the nested quotes required mode. We could even make the syntax in that mode be `to_json('...')`.